builtin/umask.c

```diff
@@ -7,26 +8,72 @@
 #include "builtin.h"
 
 /*
- * TODO: Implement -S flag and symbolic mode
+ * TODO: Implement symbolic mode
  */
 
-static const char umask_usage[] = "usage: umask [mode]\n";
+static const char umask_usage[] = "usage: umask [-S] [mode]\n";
+
+static void umask_modestring(char *string, mode_t mode) {
```

Comment:
This is a good opportunity to use array arguments here. This ensures the array is big enough.

```c
static void umask_modestring(char string[static 4], mode_t mode)
```